### PR TITLE
#43: DBTestSuite: parametrize persist_data into constructor

### LINKED ISSUE #43: DBTestSuite: parametrize persist_data into constructor
## Background
As the PR title says. Basically it's a bit limiting to have this only in configuration.

### PR DESCRIPTION
**Pull request by benedeki**
* make it easy to override data persistence in the `DBTestSuite` class, without the need to override functions.

Closes #43 

Release Notes:
- easier override of the data persistence within `DBTestSuite` instance via constructor parameter (no need to function overrides)

### CODE REVIEW

File: balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala [hunk 1]
```diff
@@ -44,7 +48,9 @@ abstract class DBTestSuite extends AnyFunSuite {
   /**
    * This is the connection info for the DB. It can be overridden in the derived classes to provide specific credentials
    */
-  protected lazy val connectionInfo: ConnectionInfo = readConnectionInfoFromConfig
+  protected lazy val connectionInfo: ConnectionInfo = {
```
Comment (miroslavpojer) [inline balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala:51]: By this change, you kept the private method `readConnectionInfoFromConfig` without usage. Is that intention?

**Review by miroslavpojer — COMMENTED** [review]
- [ ] pulled
- [ ] code review done
- [ ] run all tests

File: balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala (hunk 1 above)
Comment (benedeki) [reply balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala:51]: Thanks for noticing!

File: balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala [hunk 2]
```diff
@@ -152,7 +154,7 @@ abstract class DBTestSuite(val persistDataOverride: Option[Boolean] = None) exte
   }
 
   // private functions
```
Comment (miroslavpojer) [inline balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala:156]:
```suggestion
  // protected functions
```

Comment (miroslavpojer) [reply balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala:156]: Almost done.

**Review by miroslavpojer — COMMENTED** [review]
- [ ] pulled
- [ ] code review
- [ ] test run

File: balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala (hunk 2 above)
Comment (benedeki) [reply balta/src/main/scala/za/co/absa/db/balta/DBTestSuite.scala:156]: Obsolete

**Review by miroslavpojer — DISMISSED** [review]

**Review by miroslavpojer — APPROVED** [review]

**Pull request merged**